### ISSUE TITLE
Incorrect wave direction

### ISSUE THREAD

**Issue by phet-steele** [phetsims/bending-light#359]
**This is in the published version.** Seen on iOS 10.0.2 iPad Air 2. EDIT: the problem may be the retina display.
@ariel-phet @samreid this seems kind of important. <s>I think why we have not seen any user reports is because I do NOT see this on iOS 9.3.5, only 10.</s>

Intro and More Tools screen. The reflected light, in wave form, has its lines going the wrong way:
![img_0051](https://cloud.githubusercontent.com/assets/5863899/19576741/67ea3992-96d1-11e6-9672-0973d5e3f1e5.PNG)

Blocking the beam with the intensity sensor makes them go the correct way:
![img_0052](https://cloud.githubusercontent.com/assets/5863899/19576746/6cc68aec-96d1-11e6-8ce8-9fabcc33648b.PNG)

For phetsims/tasks/issues/682.

**Comment by samreid** [comment]
I've seen related problems like this--first thing to try will be to make the webgl canvas smaller and see if the problem goes away.

**Comment by phet-steele** [comment]
@samreid I've edited the issue to reflect this, but it is not iOS 10 specific. I checked 9.3.5 on an iPad 2 and did not see the problem, so I assumed it was only a problem for 10. But the issue IS prevalent on 9.3.5 on an iPad Air 2. These means the only commonality between sightings is the iPad model and a *retina* display. I'm willing to bet it is a problem with the retina display.

**Comment by samreid** [comment]
Unassigning until I'm devoting time to Bending Light.

**Comment by phet-steele** [comment]
Kathy's comments from #368:

> Note, this bug does not appear in webgl=false version, only webgl version.

>@jonathanolson - When there is a lull due to testing or other things you are waiting on, can you take a look at this error since it is only in WebGL code?

>It actually seems like it might be something general in WebGL node that should be corrected, and just manifests itself here for some reason.

And a comment from @ariel-phet:

>Marking at low priority since this is a "as time allows" bug and has not been reported to phethelp

**Comment by samreid** [comment]
One more note: it seems like the problem is related to the total size of the canvas, and that reducing the webgl drawable area may make the problem go away.  (mentioned inhttps://github.com/phetsims/bending-light/issues/359#issuecomment-255230720 )

**Comment by samreid** [comment]
@kathy-phet said:

> It actually seems like it might be something general in WebGL node that should be corrected, and just manifests itself here for some reason.

I have seen several occasions where having the geometry "too big" causes rendering artifacts like this.  I'm not sure how big is too big, but the problem sometimes (always?) goes away when reducing the geometry size.

**Comment by samreid** [comment]
This problem occurs on my iPhone 5S running iOS 10.2 but not on my iPad3 running 9.3.5.

**Comment by samreid** [comment]
This problem was identified for the transmitted wave in https://github.com/phetsims/bending-light/issues/147 but the solution wasn't applied to the reflected wave.  I'll apply it to the reflected wave shortly.

**Comment by samreid** [comment]
I published an RC here which includes the geometry fix and the wave color fix from @jonathanolson:
http://www.colorado.edu/physics/phet/dev/html/bending-light/1.1.4-rc.2/bending-light_en.html

I'll create a test task.

**Comment by samreid** [comment]
Task is mentioned above.  On hold until that's complete.

**Comment by kathy-phet** [comment]
Works great here. Thx, k

Sent from my iPhone

On Feb 9, 2017, at 10:00 PM, Sam Reid <notifications@github.com<mailto:notifications@github.com>> wrote:


Task is mentioned above. On hold until that's complete.

**Comment by samreid** [comment]
Ready for RC test in https://github.com/phetsims/tasks/issues/786

**Comment by phet-steele** [comment]
Wave direction was correct on (rc.3):

- iOS 10.2 retina
- iPad 2
- JB's iPhone 5c

Closing, thanks @samreid!
